src/environmentd/tests/server.rs

```diff
@@ -2054,6 +2054,78 @@ fn test_internal_console_redirect() {
     assert_eq!(redirected.load(Ordering::Relaxed), true);
 }
 
+#[mz_ore::test]
+#[cfg_attr(miri, ignore)] // too slow
+fn test_internal_http_auth() {
+    let server = util::start_server(util::Config::default()).unwrap();
+    let json = r#"{"query": "SELECT current_user;"}"#;
+    let json: serde_json::Value = serde_json::from_str(json).unwrap();
+    let url = Url::parse(&format!(
+        "http://{}/api/sql",
+        server.inner.internal_http_local_addr()
+    ))
+    .unwrap()
+    .clone();
```

Comment:
you don't need this clone but i assume clippy will catch it. i run clippy like `env CARGO_INCREMENTAL=0 cargo clippy --all-targets -- -D warnings`. you'll also want to run `cargo fmt`